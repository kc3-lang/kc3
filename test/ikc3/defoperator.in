quote %KC3.Operator{}
%KC3.Operator{}
quote def operator_muul = %KC3.Operator{sym: :****, symbol_value: cfn Tag "tag_mul" (Tag, Tag, Result), operator_precedence: 11, operator_associativity: :left}
def operator_muul = %KC3.Operator{sym: :****, symbol_value: cfn Tag "tag_mul" (Tag, Tag, Result), operator_precedence: 11, operator_associativity: :left}
quote 4 **** 4
4 **** 4
quote 4 **** 4 **** 4
4 **** 4 **** 4
quote 4 **** 4 + 4
4 **** 4 + 4
