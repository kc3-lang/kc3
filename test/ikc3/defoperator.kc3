quote %KC3.Operator{symbol_value: void}
%KC3.Operator{symbol_value: void}
quote def operator_muul = %KC3.Operator{sym: :****, symbol_value: cfn Tag "tag_mul" (Tag, Tag, Result), operator_precedence: 11, operator_associativity: :left}
def operator_muul = %KC3.Operator{sym: :****, symbol_value: cfn Tag "tag_mul" (Tag, Tag, Result), operator_precedence: 11, operator_associativity: :left}
quote 4 **** 4
4 **** 4
quote 4 **** 4 **** 4
4 **** 4 **** 4
quote 4 **** 4 + 4
4 **** 4 + 4
