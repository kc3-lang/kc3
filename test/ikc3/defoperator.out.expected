%KC3.Op{callable: fn (a, b) { a + b }}
%KC3.Op{sym: :+,
        arity: 2,
        special: false,
        precedence: 0,
        associativity: 1,
        callable: fn (a, b) { a + b }}
defoperator %KC3.Op{sym: :****,
                    precedence: 13,
                    callable: cfn Tag "tag_mul" (Tag, Tag, Result)}
%KC3.Op{sym: :****,
        arity: 2,
        special: false,
        precedence: 13,
        associativity: 1,
        callable: cfn Tag "tag_mul" (Tag, Tag, Result)}
4 **** 4
16
4 **** 4 **** 4
64
4 **** 4 + 4
20
to_lisp(quote 4 **** 4 + 4)
[+, [****, 4, 4], 4]
